dhall-docs/src/Dhall/Docs.hs

```diff
@@ -162,32 +178,57 @@ saveHtml inputAbsDir outputAbsDir packageName t@(absFile, _) = do
 
     let relativeResourcesPath = resolveRelativePath outputAbsDir htmlOutputDir
 
+    let strippedHeader = stripCommentSyntax header
+    headerAsHtml <- case markdownToHtml absFile strippedHeader of
+        Left err -> do
+            putStrLn $ markdownParseErrorAsWarning err
+            return $ Lucid.toHtml strippedHeader
+        Right html -> return html
+
     Lucid.renderToFile (Path.fromAbsFile htmlOutputFile)
-        $ filePathHeaderToHtml t DocParams {..}
+        $ filePathHeaderToHtml (absFile, headerAsHtml) DocParams {..}
 
     return htmlOutputFile
   where
     addHtmlExt :: Path b File -> IO (Path b File)
     addHtmlExt = Path.addExtension ".html"
 
+    markdownParseErrorAsWarning :: MarkdownParseError -> String
+    markdownParseErrorAsWarning MarkdownParseError{..} =
+        "\n\ESC[1;33mWarning\ESC[0m\n\n" <>
+        Text.Megaparsec.errorBundlePretty unwrap <>
+        "The original non-markdown text will be pasted in the documentation"
+
+    stripCommentSyntax :: Header -> Text
+    stripCommentSyntax (Header h)
+        | Just s <- Data.Text.stripPrefix "--" strippedHeader
+            = s
+        | Just commentPrefixStripped <- Data.Text.stripPrefix "{-" strippedHeader
+        , Just commentSuffixStripped <- Data.Text.stripSuffix "-}" commentPrefixStripped
+            = commentSuffixStripped
+        | otherwise = strippedHeader
```

Comment:
Why does this strip whitespace, while the other branches don't?

Reply:
All of the other branches start with `strippedHeader`, so all of them are stripped.

Reply:
mmm but text inside comments might be have whitespaces as well. I'll fix it